### PR TITLE
Rails 6 support

### LINKED ISSUE #184: Add support for Rails 6.0.z releases
Rails version 6.0.0 was released on 08/16/2019, and in accordance with the charter of the current phase of the Component Maintenance Working Group, the CircleCI configuration for this should be updated.

### PR DESCRIPTION
**Pull request by cjcolvar**
Builds on #192 and #201.  Supersedes #202.  Fixes #184.

Add support for Rails 6.0 and run javascript tests with jasmine-browser-runner.  Some small fixes were needed to get the javascript tests to pass.

This would probably constitute a new major version because it drops support for Rails versions < 5.2 and ruby < 2.5.
Rails 6.1 currently fails when trying to include Hydra::AccessControls::Permissions (See [this failed circleci run](https://app.circleci.com/pipelines/github/samvera/hydra-editor/105/workflows/fc0be13d-58ea-4a7a-9baf-7c86c610d62c/jobs/705/parallel-runs/0/steps/0-119))

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -19,7 +19,7 @@ jobs:
       name: 'samvera/ruby_fcrepo_solr'
       ruby_version: << parameters.ruby_version >>
     environment:
-      ENGINE_CART_RAILS_OPTIONS: --skip-git --skip-bundle --skip-listen --skip-spring --skip-yarn --skip-keeps --skip-coffee --skip-puma --skip-test
+      ENGINE_CART_RAILS_OPTIONS: --skip-git --skip-bundle --skip-listen --skip-spring --skip-keeps --skip-coffee --skip-puma --skip-test
```
Comment (jrgriffiniii) [inline .circleci/config.yml:22]: 👍

File: Rakefile [hunk 2]
```diff
@@ -30,13 +30,6 @@ require 'fcrepo_wrapper'
 require 'active_fedora/rake_support'
 require 'engine_cart/rake_task'
 
-require 'jasmine'
```
Comment (jrgriffiniii) [inline Rakefile:33]: I am sorry that this consumed so much labor, this should have been removed at a much earlier date given these problems.

File: lib/tasks/hydra-editor_tasks.rake [hunk 3]
```diff
@@ -2,3 +2,12 @@
 # task :hydra-editor do
 #   # Task goes here
 # end
+
+desc "Run rake tasks with internal test app"
+namespace :hydra_editor do
+  task recompile_js: [:environment] do
+    within_test_app do
+      system "bundle exec rake assets:clobber assets:precompile"
```
Comment (jrgriffiniii) [inline lib/tasks/hydra-editor_tasks.rake:10]: 👍

File: spec/javascripts/jasmine_spec.rb [hunk 4]
```diff
@@ -1,25 +1,20 @@
...
-# Run the jasmine tests by running the jasmine:ci rake command and parses the output for failures.
+# Run the jasmine tests by running the `npx jasmine-browser-runner` command and parses the output for failures.
 # The spec will fail if any jasmine tests fails.
 describe "Jasmine" do
   it "expects all jasmine tests to pass" do
-    load_rake_environment ["#{jasmine_path}/lib/jasmine/tasks/jasmine.rake"]
-    jasmine_out = run_task 'jasmine:ci'
-    if jasmine_out.include? "0 failures"
+    Rake::Task['hydra_editor:recompile_js'].invoke
+    jasmine_out = `npx jasmine-browser-runner runSpecs`
```
Comment (jrgriffiniii) [inline spec/javascripts/jasmine_spec.rb:8]: 👍

**Review by jrgriffiniii — APPROVED** [review]

**Comment by hackartisan** [comment]
🎉 nice work y'all thanks for getting it over the line!

**Comment by jrochkind** [comment]
Nice work! Interesting to note this is almost *all* "test setup" changes, no changes really had to be made to functionality, just to test setup.

**Pull request merged**